src/benchmarks/micro/runtime/JetStream/Segmentation/segmentation.cs

```diff
@@ -0,0 +1,491 @@
+                            cost[segmentEnd][k + 1] = (float)totalCost;
+                            previousNode[segmentEnd][k + 1] = segmentStart;
+                        }
+                    }
+                }
+            }
+        }
+
+        public class OptimalSegmentation {
+            public int[] segmentation;
+            public float cost;
+        }
+
+        public static OptimalSegmentation findOptimalSegmentation (double[] values, SampleVarianceUpperTriangularMatrix costMatrix, int segmentCount) {
+            var cost = allocateCostUpperTriangularForSegmentation(values, segmentCount);
+            var previousNode = allocatePreviousNodeForSegmentation(values, segmentCount);
+
+            findOptimalSegmentationInternal(cost, previousNode, values, costMatrix, segmentCount);
+
+            var segmentEnd = values.Length - 1;
+            var segmentation = new int[segmentCount + 1];
+            segmentation[segmentCount] = values.Length;
+
+            for (var k = segmentCount; k > 0; k--) {
+                if (segmentEnd < 0)
+                    break;
+
+                segmentEnd = previousNode[segmentEnd][k];
+                segmentation[k - 1] = segmentEnd;
+            }
+
+            var costOfOptimalSegmentation = cost[values.Length - 1][segmentCount];
+
+            return new OptimalSegmentation {segmentation = segmentation, cost = costOfOptimalSegmentation};
+        }
+    }
+
+    public class SampleVarianceUpperTriangularMatrix {
+        public float[][] costMatrix;
+
+        public SampleVarianceUpperTriangularMatrix (double[] values) {
+            // The cost of segment (i, j].
+            costMatrix = new float[values.Length - 1][];
+            for (var i = 0; i < values.Length - 1; i++) {
+                var remainingValueCount = values.Length - i - 1;
+                costMatrix[i] = new float[remainingValueCount];
+                var sum = values[i];
+                var squareSum = sum * sum;
+                costMatrix[i][0] = 0;
+                for (var j = i + 1; j < values.Length; j++) {
+                    var currentValue = values[j];
+                    sum += currentValue;
+                    squareSum += currentValue * currentValue;
+                    var sampleSize = j - i + 1;
+                    var stdev = Statistics.sampleStandardDeviation(sampleSize, sum, squareSum);
+                    costMatrix[i][j - i - 1] = (float)(stdev > 0 ? sampleSize * Math.Log(stdev * stdev) : 0);
+                }
+            }
+        }
+
+        public double costBetween (int from, int to) {
+            if (from >= this.costMatrix.Length || from == to)
+                return 0; // The cost of the segment that starts at the last data point is 0.
+            return this.costMatrix[from][to - from - 1];
+        }
+    }
+
+    [BenchmarkCategory(Categories.Runtime, Categories.JetStream)]
```

Comment:
I am sorry, I've missed this file in my previous review

```suggestion
    [BenchmarkCategory(Categories.Runtime)]
```